sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -40,6 +40,23 @@ namespace Azure { namespace Core { namespace Http {
       left.insert(right.begin(), right.end());
       return left;
     }
+
+    /**
+     * @brief Inserts a header into \p headers checking that \p headerName contains valid characters
+     *
+     * @param headers The headers map where to insert header
+     * @param headerName The header name for the header to be inserted.
+     * @param headerValue The header value for the header to be inserted.
+     * @param invalidHeaderIndex optional output pointer to a size_t that refers to the invalid
+     * index when this method fails.
+     * @return true when header is successfully added.
```

Comment:
Same comment as below, you need single `@return section`, backtick true and false.